src/library_webgl.js

```diff
@@ -626,7 +626,8 @@ var LibraryGL = {
       if (Module['preinitializedWebGLContext']) {
         var ctx = Module['preinitializedWebGLContext'];
 #if MAX_WEBGL_VERSION >= 2
-        webGLContextAttributes.majorVersion = (typeof WebGL2RenderingContext != 'undefined' && ctx instanceof WebGL2RenderingContext) ? 2 : 1;
+        // The ctx object may not be of a known class (e.g. it may be a debug wrapper), so we ask it for its version instead.
+        webGLContextAttributes.majorVersion = ctx.getParameter(ctx.VERSION).match(/^WebGL (\d+).(\d+)/)[1];
```

Comment:
How about `(/^WebGL \d+.(\d+)/)[0];`  ?

Reply:
That doesn't seem right.

```
> "WebGL 2.0".match(/^WebGL (\d+).(\d+)/)[1]
'2'
> "WebGL 2.0".match(/^WebGL \d+.(\d+)/)[0];
'WebGL 2.0'
```

We want the former result. And I think I should actually wrap it in `Number()`. But I can remove the second capturing group, if that's what you meant.

Reply:
Ah yes, thanks!